references/classification/WildFire/torch/train.py

```diff
@@ -0,0 +1,310 @@
+    with torch.no_grad():
+        for x, target in test_loader:
+
+            target = target.type(torch.LongTensor).squeeze()
+            # Work with tensors on GPU
+            if device.startswith('cuda'):
+                x, target = x.cuda(), target.cuda()
+
+            # Forward + Backward & optimize
+            outputs = model.forward(x)
+            val_loss += criterion(outputs, target).item()
+            # Index of max log-probability
+            pred = outputs.max(1, keepdim=True)[1]
+            correct += pred.eq(target.view_as(pred)).sum().item()
+            targets += x.size(0)
+    val_loss /= len(test_loader)
+    acc = correct / targets
+
+    return val_loss, acc
+
+
+def main(args):
+
+    if args.deterministic:
+        set_seed(42)
+
+    # Set device
+    if args.device is None:
+        if torch.cuda.is_available():
+            args.device = 'cuda:0'
+        else:
+            args.device = 'cpu'
+
+    #Create Dataloaders
+
+    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+
+    train_transforms = transforms.Compose([
+        transforms.RandomResizedCrop(size=args.resize, scale=(0.8, 1.0)),
+        transforms.RandomRotation(degrees=15),
+        transforms.ColorJitter(),
+        transforms.RandomHorizontalFlip(),
+        transforms.ToTensor(),
+        normalize
+    ])
+
+    val_transforms = transforms.Compose([
+        transforms.Resize(size=args.resize),
+        transforms.CenterCrop(size=args.resize),
+        transforms.ToTensor(),
+        normalize
+    ])
+
+    tf = {'train': train_transforms, 'test': val_transforms, 'val': val_transforms}
+
+    metadata = pd.read_csv(args.metadata)
+
+    wildfire = WildFireDataset(metadata=metadata, path_to_frames=Path(args.DB), target_names=['fire'])
+
+    ratios = {'train': args.ratio_train, 'val': args.ratio_val, 'test': args.ratio_test}
+
+    splitter = WildFireSplitter(ratios, transforms=tf)
+    splitter.fit(wildfire)
+
+    train_loader = DataLoader(splitter.train, batch_size=args.batch_size, shuffle=True)
+    val_loader = DataLoader(splitter.val, batch_size=args.batch_size, shuffle=True)
+
+    # Model definition
+    model = models.__dict__[args.model](imagenet_pretrained=args.pretrained,
+                                        num_classes=args.nb_class, lin_features=args.lin_feats,
+                                        concat_pool=args.concat_pool, bn_final=args.bn_final,
+                                        dropout_prob=args.dropout_prob)
+
+    # Freeze layers
+    if not args.unfreeze:
+        # Model is sequential
+        for p in model[1].parameters():
+            p.requires_grad = False
+
+    # Resume
+    if args.resume:
+        model.load_state_dict(torch.load(args.resume)['model'])
+
+    # Send to device
+    model.to(args.device)
+
+    # Loss function
+    criterion = nn.CrossEntropyLoss()
+
+    # optimizer
+    optimizer = optim.Adam(model.parameters(),
```

Comment:
I would recommend to let the ability to use other Optimizers than Adam

Reply:
Agreed, but perhaps we can introduce this in another PR?

Reply:
Yes let's do that